UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/FileContentHelper.cs

```diff
@@ -1,58 +1,92 @@
-using System.Threading.Tasks;
-using UKHO.ExchangeSetService.API.FunctionalTests.Models;
-using System.Net.Http;
-using UKHO.PeriodicOutputService.API.FunctionalTests.Helpers;
-using UKHO.PeriodicOutputService.API.FunctionalTests.Models;
+﻿using UKHO.PeriodicOutputService.API.FunctionalTests.Helpers;
 
 namespace UKHO.ExchangeSetService.API.FunctionalTests.Helpers
 {
     public static class FileContentHelper
     {
         private static TestConfiguration Config = new TestConfiguration();
-        private static FssApiClient FssApiClient = new FssApiClient();
-       
 
-        public static async Task<List<string>> CreateExchangeSetFileForLargeMedia(HttpResponseMessage apiEssResponse, string FssJwtToken, string BatchId)
+        public static async Task<List<string>> CreateExchangeSetFileForLargeMedia(string BatchId, string FssJwtToken)
         {
             List<string> downloadFolderPath = new List<string>();
-            Assert.That((int)apiEssResponse.StatusCode, Is.EqualTo(200), $"Incorrect status code is returned {apiEssResponse.StatusCode}, instead of the expected status 200.");
 
-           for (int mediaNumber = 1; mediaNumber <= 2; mediaNumber++)
-           {
-                var FolderName = $"M0{mediaNumber}X02";
-                var downloadFileUrl = $"{Config.FssConfig.BaseUrl}/batch/{BatchId}/files/{FolderName}.zip";
+            for (int mediaNumber = 1; mediaNumber <= 2; mediaNumber++)
+            {
+                var folderName = $"M0{mediaNumber}X02.zip";
+                var downloadFileUrl = $"{Config.FssConfig.BaseUrl}/batch/{BatchId}/files/{folderName}";
 
-                var DownloadedFolder = await FssBatchHelper.DownloadedFolderForLargeFiles(downloadFileUrl, FssJwtToken, FolderName);
+                var downloadedFolder = await FssBatchHelper.DownloadedFolderForLargeFiles(downloadFileUrl, FssJwtToken, folderName);
 
-                downloadFolderPath.Add(DownloadedFolder);
-           }
+                downloadFolderPath.Add(downloadedFolder);
+            }
             return downloadFolderPath;
         }
 
-        public static void DeleteDirectory(string fileName)
+        public static void DeleteZipIsoSha1Files(string fullFileName)
         {
-            if (File.Exists(fileName))
-            {
-                string folder = Path.GetFileName(fileName);
-                if (folder.Contains(".zip"))
-                {
-                    folder = folder.Replace(".zip", "");
-                }
+            string path = Path.GetTempPath();
 
-                //Delete V01X01.zip/M01XO2.zip/M02XO2.zip file from temp Directory
-                if (File.Exists(fileName))
-                {
-                    File.Delete(fileName);
-                }
+            if (Directory.Exists(Path.Combine(path, fullFileName)))
+            {
+                Directory.Delete(Path.Combine(path, fullFileName),true);
             }
+            else if (File.Exists(Path.Combine(path, fullFileName + ".zip")))
```

Comment:
Hi Jason, as per our expectation we need to delete .zip , .iso , .iso.sha1 and extracted folders from temp folder but if we implement the above solution it just deletes the extracted folders(M01X02 & M02X02). The other files remain in the temp folder.
![image](https://user-images.githubusercontent.com/85937324/186148000-63c8ae30-e0c0-4a23-8494-95f1217ee387.png)

Thanks.

Reply:
Hi Arit - but the existing code doesn't do that because of the **else** parts of the if statement. It will only ever delete the directory, and will only delete the files if the directory doesn't exist. To do what you want you would need the following:

            if (Directory.Exists(Path.Combine(path, fullFileName)))
            {
                Directory.Delete(Path.Combine(path, fullFileName),true);
            }

            File.Delete(Path.Combine(path, fullFileName + ".zip"));
            File.Delete(Path.Combine(path, fullFileName + ".iso"));
            File.Delete(Path.Combine(path, fullFileName + ".iso.sha1"));
            File.Delete(Path.Combine(path, fullFileName));

Only the above will remove the directory AND remove the files. Is that what you want to do or have I misunderstood?

Reply:
Code updated. Thanks for the help.